ws-server/src/server.rs

```diff
@@ -301,21 +306,23 @@ async fn background_task(
 	// Send results back to the client.
 	tokio::spawn(async move {
 		while !stop_server2.shutdown_requested() {
-			match rx.next().await {
-				Some(response) => {
-					// If websocket message send fail then terminate the connection.
-					if let Err(err) = send_ws_message(&mut sender, response).await {
-						tracing::error!("WS transport error: {:?}; terminate connection", err);
-						break;
-					}
+			if let Some(response) = rx.next().await {
+				// If websocket message send fail then terminate the connection.
+				if let Err(err) = send_ws_message(&mut sender, response).await {
+					tracing::error!("WS transport error: {:?}; terminate connection", err);
+					break;
 				}
-				None => break,
-			};
+			} else {
+				break;
+			}
 		}
-		// terminate connection.
+
+		// Terminate connection and send close message.
 		let _ = sender.close().await;
-		// NOTE(niklasad1): when the receiver is dropped no further requests or subscriptions
-		// will be possible.
+
+		// Force `conn_tx` to this async block and close it down
+		// when the connection closes to be on safe side.
```

Comment:
//cc @dvdplm this is what I meant by closing down the channel.

once this tasks terminates we drop `conn_tx` before I dropped it close/drop shouldn't matter just that it's moved into this block.

Reply:
ok, ty